### ISSUE TITLE
[routing] Sayreville, NJ to South River. NJ Issue

### ISSUE THREAD

**Issue by DicePenguin** [organicmaps/organicmaps#3207]
The route seems to prefer taking Main Street in Sayreville, making a right onto Washington Road to get to South River, NJ instead of the more optimal route from Main Street then making a slight right onto Macarthur Avenue then another slight right onto Washington Road which becomes Main Street in South River, NJ. The route in question can be recreated by using the coordinates provided below.

Routing Coordinates:
From: (40.475946, -74.318432) Parking Lot
To:     (40.451317, -74.381243) South River Police Department

**Comment by biodranik** [comment]
How does OpenStreetMap.org route it?

**Comment by DicePenguin** [comment]
I tested the route navigation in OpenStreetMap and it looks like it is also routing it the same. I suppose, the core of the issue lies with them?

**Comment by biodranik** [comment]
Please check the roads and their tags in OpenStreetMap. I assume that some information is outdated there or does not reflect reality. Or maybe your "optimal" criteria is wrong :)

**Comment by DicePenguin** [comment]
I've tried many methods in openstreetmap with the tags to no success. And locally, everyone in town knows if you want to get to South River, the best way is to take Macarthur Avenue. 
Reason 1: The traffic light at Main Street and Washington Road has a "delayed green" light for traffic entering from Main Street. 
Reason 2: Macarthur Avenue is a newer road and was built for the sole purpose for traveling to South River easier. 
Reason 3: The light at Macarthur Avenue as a separate way for right turns, allowing traffic to South River even on red lights because of the "right on red" allowance in the state.

**Comment by DicePenguin** [comment]
I've added a picture to further accentuate the situation. The current route in blue while red is the suggested route. Notice how it cuts the corner for a more direct route to South River.
![IMG-6016](https://user-images.githubusercontent.com/84756738/186800940-901cbb58-4c0b-44d8-b6b8-2a27a572b7f7.PNG)
.

**Comment by biodranik** [comment]
1. Are speed limits set properly for all roads?
2. OM does not add a penalty now for traffic lights and turns, we'll add it soon. If it is the reason (e.g. two routes are almost equal otherwise), then your issue will be fixed.

**Comment by DicePenguin** [comment]
Yes, I've made sure to set the speed limits for each road. Thank you for your assistance in with the issue.

**Comment by TheAdventurer64** [comment]
> Yes, I've made sure to set the speed limits for each road. Thank you for your assistance in with the issue.

Can this issue be closed?

**Comment by DicePenguin** [comment]
I don't think it can be closed because the change for penalty for traffic lights and turns have not been implemented yet. Unless I'm mistaken about that fact.
